app/Domains/RabbitMQ/Jobs/MakeInsightsJob.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Domains\RabbitMQ\Jobs;
+
+use App\Data\Collections\MediaCollection;
+use App\Data\Models\Account;
+use App\Data\Models\Insights;
+use Lucid\Units\Job;
+
+class MakeInsightsJob extends Job
+{
+    private array $message;
+
+    private Account $account;
+
+    private MediaCollection $medias;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param array $message
+     * @param Account $account
+     * @param MediaCollection $medias
+     */
+    public function __construct(array $message, Account $account, MediaCollection $medias)
+    {
+        $this->message = $message;
+        $this->account = $account;
+        $this->medias = $medias;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle(): Insights
+    {
+        return Insights::makeFromQueueMessage((int)$this->message['fetched_at'], $this->account, $this->medias);
```

Comment:
avoid passing messages, it builds tight coupling between classes (job class and `Insights` now have to know about the queue and its message structure). Break that dependency by specifying what the job needs, and changing the method name `makeFromQueueMessage` to something more adequate.